app/controllers/tag_groups_controller.rb

```diff
@@ -92,7 +92,10 @@ def search
 
     matches =
       matches.order("name").limit(
-        fetch_limit_from_params(default: 5, max: SiteSetting.max_tag_search_results),
+        fetch_limit_from_params(
+          default: SiteSetting.max_tag_search_results,
+          max: SiteSetting.max_tag_search_results,
+        ),
```

Comment:
```suggestion
        SiteSetting.max_tag_search_results.to_i
```
I think that in that case, we can completely skip `fetch_limit_from_params` method. I checked that this method ensures that the value is an integer and checks if it is between the min and max parameters. We don't need to check it anymore. I added `.to_i` for sanity, just in case there is a weird value in the site setting

Reply:
I think removing the `fetch_limit_from_params` call would be considered a breaking change for this API endpoint given that the `limit` param would do nothing for API consumers that currently consume this endpoint.

I've added a max limit to both the method call here and in the setting definition in the yaml file.